src/commonMain/kotlin/io/konform/validation/constraints/NumberConstraints.kt

```diff
@@ -13,26 +14,34 @@ public fun <T : Number> ValidationBuilder<T>.multipleOf(factor: Number): Constra
     }
 }
 
```

Comment:
:+1:

Can you add overloads for `Long` too?  Float seems like a more niche use-case so feel free to remove, keeping them is fine now you made them.

```suggestion
// The below functions are overloads of the Comparable<T> functions, provided for backwards
// compatibility and convenience for the common case where a number is constrained by an Int
```

Reply:
> Can you add overloads for `Long` too? Float seems like a more niche use-case so feel free to remove, keeping them is fine now you made them.

Kotlin automatically converts the literal to a `Long` value depending on the context:
https://github.com/konform-kt/konform/blob/8d30d468181a16eb68acb6f1ba8a6219c1e0a42b/src/commonTest/kotlin/io/konform/validation/constraints/ConstraintsTest.kt#L154-L157

That's why I was thinking an explicit overload might not be needed. I haven't tried with a value explicitly typed as `Int` though. I will check that and can certainly add them.

Reply:
ah I see, looked at the widening rules but integer literal to long does happen automatically, so shouldn't be needed

Reply:
Ah, sorry. Saw your message a moment too late and just pushed an update adding the overload for `Long`. 🙈

I checked the case where the limit value is explicitly typed, e.g., when is provided using a constant value from elsewhere:

```kotlin
val MAX: Int = 100
val validation = Validation<Long> {
    maximum(MAX)
}
```

In this case the overload still might be useful.